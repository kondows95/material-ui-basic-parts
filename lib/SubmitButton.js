"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var react_intl_1 = require("react-intl");
var MyButton_1 = __importDefault(require("./MyButton"));
var SubmitButton = function (props) {
    var formatMessage = react_intl_1.useIntl().formatMessage;
    var copyProps = {};
    for (var _i = 0, _a = Object.keys(props); _i < _a.length; _i++) {
        var key = _a[_i];
        if (key !== 'loading') {
            copyProps[key] = props[key];
        }
    }
    var msgId = 'MUBP.SubmitButton';
    return (react_1.default.createElement(MyButton_1.default, __assign({}, copyProps, { label: formatMessage({ id: msgId, defaultMessage: 'Submit' }), loading: props.loading ? true : false, type: "submit", color: "primary", fullWidth: true })));
};
exports.default = SubmitButton;
