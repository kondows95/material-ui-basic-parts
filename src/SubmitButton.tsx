import React, { PropsWithChildren } from 'react';
import { Button, Box, CircularProgress, Theme } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { useIntl } from 'react-intl';

const useStyles = makeStyles((theme: Theme) => ({
    progress: {
        position: 'absolute',
        top: '50%',
        left: '50%',
        marginTop: -9,
        marginLeft: -12,
    },
    item: {
        [theme.breakpoints.down('xs')]: {
            padding: theme.spacing(2, 0, 1), //TXB
        },
        [theme.breakpoints.up('sm')]: {
            padding: theme.spacing(1, 0, 1, 1), //TRBL
        },
    },
}));

interface Props {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    [key: string]: any;
    loading?: boolean;
}

const SubmitButton: React.FC<PropsWithChildren<Props>> = (props) => {
    const classes = useStyles();
    const { formatMessage } = useIntl();

    const loading = props.loading ? true : false;
    const myProp: Props = {};
    for (const key of Object.keys(props)) {
        if (key !== 'loading') {
            myProp[key] = props[key];
        }
    }

    let progress = null;
    if (loading) {
        progress = <CircularProgress size={20} color="primary" className={classes.progress} />;
    }

    return (
        <Box className={classes.item}>
            <Button
                {...myProp}
                disabled={loading}
                variant="contained"
                type="submit"
                color="primary"
                fullWidth
                data-testid="submit-button"
            >
                {formatMessage({ id: 'MUIBP.submit', defaultMessage: 'Submit' })}
                {progress}
            </Button>
        </Box>
    );
};
export default SubmitButton;
