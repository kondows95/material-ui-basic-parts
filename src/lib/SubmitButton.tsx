import React from 'react';
import { useIntl } from 'react-intl';
import MyButton from './MyButton';

interface Props {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    [key: string]: any;
    loading?: boolean;
}

interface CopyProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    [key: string]: any;
}

const SubmitButton: React.FC<Props> = (props) => {
    const { formatMessage } = useIntl();

    const copyProps: CopyProps = {};
    for (const key of Object.keys(props)) {
        if (key !== 'loading') {
            copyProps[key] = props[key];
        }
    }

    const msgId = 'MUBP.SubmitButton';
    return (
        <MyButton
            {...copyProps}
            label={formatMessage({ id: msgId, defaultMessage: 'Submit' })}
            loading={props.loading ? true : false}
            type="submit"
            color="primary"
            fullWidth
        />
    );
};
export default SubmitButton;
