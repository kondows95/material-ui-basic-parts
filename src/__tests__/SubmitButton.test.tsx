import React from 'react';
import TestLib from 'react-testing-library-wrapper';
import { IntlProvider } from 'react-intl';
import SubmitButton from '../lib/SubmitButton';

let tLib: TestLib;

describe('Basic Usage', () => {
    beforeEach(() => {
        tLib = new TestLib(
            (
                <IntlProvider locale="en">
                    <SubmitButton loading={true} className="MyClass" data-testid="myId" />
                </IntlProvider>
            )
        );
    });

    it('Label is Submit', () => {
        expect(tLib.get('myId').textContent).toBe('Submit');
    });

    it('You can disable button and show CircularProgress', () => {
        const elem = tLib.get('myId') as HTMLButtonElement;
        expect(elem.disabled).toBeTruthy();
        expect(elem.innerHTML).toContain('CircularProgress');
    });

    it('You can add className', () => {
        expect(tLib.get('myId').getAttribute('class')).toContain('MyClass');
    });

    it('Snapshot', () => {
        expect(tLib.render.asFragment()).toMatchSnapshot();
    });
});

it('Check Default Values', () => {
    tLib = new TestLib(
        (
            <IntlProvider locale="en">
                <SubmitButton data-testid="myId" />
            </IntlProvider>
        )
    );
    const elem = tLib.get('myId') as HTMLButtonElement;
    expect(elem.disabled).toBeFalsy();
    expect(elem.innerHTML).not.toContain('CircularProgress');
});
