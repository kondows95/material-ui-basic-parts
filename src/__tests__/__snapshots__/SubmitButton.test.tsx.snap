// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Usage Snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MyClass MuiButton-containedPrimary Mui-disabled MuiButton-fullWidth Mui-disabled"
    data-testid="myId"
    disabled=""
    tabindex="-1"
    type="submit"
  >
    <span
      class="MuiButton-label"
    >
      Submit
      <div
        class="MuiCircularProgress-root makeStyles-progress-136 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 20px; height: 20px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </span>
  </button>
</DocumentFragment>
`;
